src/legacy/index.ts

```diff
@@ -263,8 +285,23 @@ async function buildSubtree(
     return depTree;
   }
 
+  if (maybeDeduplicationSet) {
```

Comment:
Can we reduce the nesting here and just check both in the same line?

Reply:
no, because there's a logic to be executed when only one or two of conditions are valid.